content/docs/learning-guide/tutorials/rendering/vegetation-bending-tutorial.md

```diff
@@ -0,0 +1,704 @@
+
+Now, we have a tree (at an offset)! This tree is important because it uses vertex colors that we will be using to determine how the tree should bend.
+
+Note on vertex colors: You can use a DCC tool to color the vertices to match the type of bending you want to do on each part of the tree: 
+* Red vertex color: smaller movement with high frequency of random sinusoidal noise.
+* Green vertex color: delays the start of the movement for variations with high frequency of random sinusoidal noise.
+* Blue vertex color: larger movement and bending with low frequency of sinusoidal noise. 
+In the case of our tree, the trunk's vertices are colored blue and leaves are colored red.
+
+### Add a shader option
+Our tree mesh that we just added has colored vertices; however, not every mesh that you want to apply your material type to may have colored vertices. Therefore, we want to add a [*shader option*](/docs/atom-guide/dev-guide/shaders/azsl/#shader-variant-options) that allows the vertex shader to handle both of these cases. We also want to add the reference to the colors so we can use them in the code.
+
+1. Open `{your-project-path}\Materials\Types\VegetationBending_Common.azsli`.
+1. At the bottom of the file, add:
+
+   ```
+   option bool o_color_isBound;
+   ```
+   We place this variable in the common file because we will be using it in all the passes.
+1. Open `{your-project-path}\Materials\Types\VegetationBending_ForwardPass.azsli`. Inside `struct VegetationVSInput`, add another field:
+
+   ```
+   float4 m_optional_color : COLOR0;
+   ```
+   If the material's vertices are colored, `m_optional_color` will be set at runtime if it's available. Then, if `m_optional_color` is available, a soft naming convention will set `o_color_isBound` to true, which we can use later to determine if we want to perform the bending or not. Note that this soft naming convention is a very specific sub-feature of shader options that are set based on the presence or absence of an optional vertex stream, as opposed to shader options set based on material properties.
+   
+   All of the fields are indicated by [HLSL semantics](https://docs.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-semantics#vertex-shader-semantics). The engine processes the semantics and updates the fields accordingly.
+1. Inside the function `VegetationBending_ForwardPassVS`, encase the offset code with a conditional using the option boolean:
+
+   ```
+   if (o_color_isBound) {
+      worldPosition.x += MaterialSrg::m_xOffset;
+      worldPosition.y += MaterialSrg::m_yOffset;
+   }
+   ```
+1. Repeat steps 3 and 4 with the depth pass in `{your-project-path}\Materials\Types\VegetationBending_DepthPass.azsli`.
+1. Save both files and open your level in the **Editor** from the previous steps. You should see that the tree entity is offset, but the shader ball (which doesn't have a vertex color stream) is not!
```

Comment:
```suggestion
1. Save both files and open your level in the **Editor** from the previous steps. 

You should see that the tree entity is offset, but the shader ball is not! That's because the shader ball doesn't have a vertex color stream.
```